cmake/farm_ng_module.cmake

```diff
@@ -5,57 +5,57 @@ include(CMakePackageConfigHelpers)
 
 macro(farm_ng_module)
 
-  message(STATUS "farm_ng_module ${PROJECT_NAME}")
-  # Configure CCache if available
-  find_program(CCACHE_FOUND ccache)
-  if(CCACHE_FOUND)
-    set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ccache)
-    set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
-  endif(CCACHE_FOUND)
+  # message(STATUS "farm_ng_module ${PROJECT_NAME}")
```

Comment:
Unused now? How will this affect downstream users?

Reply:
Not intending to merge like this, this PR is just a checkpoint of an experiment for discussion. 

One part of the experiment was refactoring these macros. So far in the experiment I think I've managed to factor out all the functionality of this macro into more appropriate places. But yes, if we were to merge like this it would break downstream repositories that rely on `farm_ng_module`.